thevoid/server.cpp

```diff
@@ -50,9 +50,62 @@
 namespace ioremap {
 namespace thevoid {
 
-class server_data;
+namespace signal_handler {
 
-static std::weak_ptr<signal_handler> global_signal_set;
+std::mutex lock;
+std::set<base_server*> all_servers;
+
+static
+void stop(int signal_value)
+{
+	std::lock_guard<std::mutex> locker(lock);
```

Comment:
Having a bug is not an excuse to create another one.
You should fix bugs, not copy them around.